test/integration-test/TestIIRDeserializer.cpp

```diff
@@ -0,0 +1,607 @@
+  lhs->setID(target->nextUID());
+  auto rhs = std::make_shared<ast::FieldAccessExpr>(sirInField->Name);
+  rhs->setID(target->nextUID());
+
+  int in_fieldID = target->getMetaData().addField(iir::FieldAccessType::FAT_APIField,
+                                                     sirInField->Name, sirInField->fieldDimensions);
+  int out_fieldID = target->getMetaData().addField(
+      iir::FieldAccessType::FAT_APIField, sirOutField->Name, sirOutField->fieldDimensions);
+
+  auto expr = std::make_shared<ast::AssignmentExpr>(lhs, rhs);
+  expr->setID(target->nextUID());
+  auto stmt = std::make_shared<ast::ExprStmt>(expr);
+  stmt->setID(target->nextUID());
+  auto statement = std::make_shared<Statement>(stmt, nullptr);
+  auto insertee = make_unique<iir::StatementAccessesPair>(statement);
+  // Add the accesses to the Pair:
+  std::shared_ptr<iir::Accesses> callerAccesses = std::make_shared<iir::Accesses>();
+  callerAccesses->addWriteExtent(out_fieldID, iir::Extents{0, 0, 0, 0, 0, 0});
+  callerAccesses->addReadExtent(in_fieldID, iir::Extents{0, 0, 0, 0, 0, 0});
+  insertee->setCallerAccesses(callerAccesses);
+  // And add the StmtAccesspair to it
+  IIRDoMethod->insertChild(std::move(insertee));
+  IIRDoMethod->updateLevel();
+
+  // Add the control flow descriptor to the IIR
+  auto stencilCall = std::make_shared<ast::StencilCall>("generatedDriver");
+  stencilCall->Args.push_back(sirInField->Name);
+  stencilCall->Args.push_back(sirOutField->Name);
+  auto placeholderStencil = std::make_shared<ast::StencilCall>(
+      iir::InstantiationHelper::makeStencilCallCodeGenName(stencilID));
+  auto stencilCallDeclStmt = std::make_shared<iir::StencilCallDeclStmt>(placeholderStencil);
+  // Register the call and set it as a replacement for the next vertical region
+  target->getMetaData().addStencilCallStmt(stencilCallDeclStmt, stencilID);
+
+  auto stencilCallStatement = std::make_shared<Statement>(stencilCallDeclStmt, nullptr);
+  target->getIIR()->getControlFlowDescriptor().insertStmt(stencilCallStatement);
+
+  ///////////////// Generation of the Metadata
+
+  target->getMetaData().insertExprToAccessID(lhs, out_fieldID);
+  target->getMetaData().insertExprToAccessID(rhs, in_fieldID);
+  target->getMetaData().setStencilname("generated");
+
+  for(const auto& MS : iterateIIROver<iir::MultiStage>(*(target->getIIR()))) {
+    MS->update(iir::NodeUpdateType::levelAndTreeAbove);
+  }
+  // Iterate all statements (top -> bottom)
+  for(const auto& stagePtr : iterateIIROver<iir::Stage>(*(target->getIIR()))) {
+    iir::Stage& stage = *stagePtr;
+    for(const auto& doMethod : stage.getChildren()) {
+      doMethod->update(iir::NodeUpdateType::level);
+    }
+    stage.update(iir::NodeUpdateType::level);
+  }
+  for(const auto& MSPtr : iterateIIROver<iir::Stage>(*(target->getIIR()))) {
+    MSPtr->update(iir::NodeUpdateType::levelAndTreeAbove);
+  }
+}
+
+static void createLapStencilIIRInMemory(std::shared_ptr<iir::StencilInstantiation>& target) {
```

Comment:
This part should hopefullly be much simpler now (if everything works as expected) with the IIRBuilder

Reply:
Right, however, if I remember right we agreed at the sprint meeting yesterday that this issue was to be closed asap with the option to improve upon it later. Maybe we should put additional tasks w.r.t that into the backlog?